src/shared/__tests__/math.spec.ts

```diff
@@ -0,0 +1,72 @@
+import * as math from '../math'
+import {describe, it, expect} from 'vitest'
+
+const large = 9007199254740993n
+
+describe('safeNumber', () => {
+    it('leaves small numbers alone', () => {
+        expect(math.safeNumber(7)).toBe(7)
+    })
+    it('makes small bigints into numbers', () => {
+        expect(math.safeNumber(-11n)).toBe(-11)
+    })
+    it('throws on numbers of great magnitude', () => {
+        expect(() => math.safeNumber(10e40)).toThrowError('Attempt')
+    })
+    it('throws on bigints of great magnitude', () => {
+        expect(() => math.safeNumber(-large)).toThrowError('Attempt')
+    })
+})
+
+describe('floorSqrt', () => {
+    it('does nothing on negative bigints', () => {
+        expect(math.floorSqrt(-13n)).toBe(-13n)
```

Comment:
What is the rationale for returning the same thing here?

`Math.sqrt(-13)` returns `NaN`.

Reply:
I read the implementation. I see this is just how bigint-isqrt does things.

Do you understand why it doesn't do anything to negative numbers instead of returning `NaN` or something along those lines?